pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -291,10 +295,14 @@ func legacyDeviceSetPVCID(deviceSetName string, setIndex int) string {
 
 // This is the new function that generates the labels
 // It includes the pvcTemplateName in it
-func deviceSetPVCID(deviceSetName, pvcTemplateName string, setIndex int) string {
+func deviceSetPVCID(deviceSetName, pvcTemplateName string, setIndex int) (string, error) {
 	cleanName := strings.Replace(pvcTemplateName, " ", "-", -1)
 	deviceSetName = strings.Replace(deviceSetName, ".", "-", -1)
-	return fmt.Sprintf("%s-%s-%d", deviceSetName, cleanName, setIndex)
+	pvcID := fmt.Sprintf("%s-%s-%d", deviceSetName, cleanName, setIndex)
+	if len(pvcID) > 62 {
+		return "", fmt.Errorf("length of generate pvc should be less the 63 as the 'volumeMount[*].Name' has limit of 63 char. Current length'  %d", len(pvcID))
```

Comment:
How about this message so it is clear what they need to change?
```suggestion
		return "", fmt.Errorf("the OSD PVC name requested is %q (length %d) is too long and must be less than 63 characters, shorten either the storageClassDeviceSet name or the storage class name", pvcID, len(pvcID))
```

Reply:
done